src/libawkward/io/json.cpp

```diff
@@ -772,7 +772,7 @@ namespace awkward {
   };
 
   template<typename HANDLER, typename STREAM>
-  const ContentPtr
+  const std::pair<int, const BuilderPtr>
   do_parse(HANDLER& handler, rj::Reader& reader, STREAM& stream) {
```

Comment:
A Builder knows its own `length()`, so I think this extra integer is not necessary, which would simplify these signatures quite a lot. Also, we don't use bare `int` in Awkward C++; we always want to know the integer size (and integers outside of arrays are always `int64_t`).

Reply:
Currently the integer is needed because it is an indicator of multiple JSON streams for the following cases: `tests/test_0437-stream-of-many-json-files.py`

Reply:
From what I saw of the code, I thought the integer was the length of the Builder. If so, `Builder::length` can produce that on demand.

If it's not the length, what is it? The JSON streams, or "JSONS" or "[ndjson](http://ndjson.org/)" would produce a Builder with a `length` just as the "one big JSON array" case would. Is the integer 0 for non-stream, 1 for stream, or something like that?

Reply:
yes, it counts the number of JSON streams. There is an extra wrapping of the result in a list since it's not known in advance if it's going to be one or many JSON fragments.

Reply:
When the "json" versus "ndjson" interfaces are well separated, the number of JSON streams is something that only applies to the "ndjson" case, and then it _is_ the `ArrayBuilder::length`. So in the final form of this PR (separating those interfaces is a goal of this PR, right?), the extra integer won't be necessary.

All we need to know at the end is the length of the array that is produced (`ak.from_buffers` needs it), not the lengths of any substructure within that.